Plugin/Magento/Config/Model/ConfigPlugin.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2017-2022 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+namespace Bolt\Boltpay\Plugin\Magento\Config\Model;
+
+use Bolt\Boltpay\Api\StoreConfigurationManagerInterface;
+use Bolt\Boltpay\Helper\Config as BoltConfig;
+use Magento\Config\Model\Config;
+use Magento\Store\Model\ScopeInterface;
+use Magento\Store\Api\WebsiteRepositoryInterface;
+use Magento\Store\Api\StoreRepositoryInterface;
+use Magento\Framework\App\ScopeInterface as AppScopeInterface;
+
+/**
+ * Send bolt request after configuration save
+ */
+class ConfigPlugin
+{
+    /**
+     * @var StoreConfigurationManagerInterface
+     */
+    private $storeConfigurationManager;
+
+    /**
+     * @var BoltConfig
+     */
+    private $boltConfig;
+
+    /**
+     * @var WebsiteRepositoryInterface
+     */
+    private $websiteRepository;
+
+    /**
+     * @var StoreRepositoryInterface
+     */
+    private $storeRepository;
+
+    /**
+     * @param StoreConfigurationManagerInterface $storeConfigurationManager
+     * @param BoltConfig $boltConfig
+     * @param WebsiteRepositoryInterface $websiteRepository
+     * @param StoreRepositoryInterface $storeRepository
+     */
+    public function __construct(
+        StoreConfigurationManagerInterface $storeConfigurationManager,
+        BoltConfig $boltConfig,
+        WebsiteRepositoryInterface $websiteRepository,
+        StoreRepositoryInterface $storeRepository
+    ) {
+        $this->storeConfigurationManager = $storeConfigurationManager;
+        $this->boltConfig = $boltConfig;
+        $this->websiteRepository = $websiteRepository;
+        $this->storeRepository = $storeRepository;
+    }
+
+    /**
+     * Send request to bolt after magento configuration update
+     *
+     * @param Config $subject
+     * @param $result
+     * @return Config
+     */
+    public function afterSave(
+        Config $subject,
+        Config $result
+    ): Config {
+
+        if ($result->getScope() == AppScopeInterface::SCOPE_DEFAULT) {
```

Comment:
Let's introduce a feature switch enabled by default and early return if the feature switch is disabled.
We always add ways to remotely disable functionality in case if we have any bugs in code

Reply:
Feature switch is introduced let's add check here